lib/ipam/ipam.go

```diff
@@ -1325,12 +1330,23 @@ func (c ipamClient) ReleaseByHandle(ctx context.Context, handleID string) error
 	}
 	handle := allocationHandle{obj.Value.(*model.IPAMHandle)}
 
-	for blockStr, _ := range handle.Block {
+	for blockStr := range handle.Block {
 		_, blockCIDR, _ := net.ParseCIDR(blockStr)
 		if err := c.releaseByHandle(ctx, handleID, *blockCIDR); err != nil {
 			return err
 		}
 	}
+
```

Comment:
Found this while testing out various weird scenarios (e.g., deleting some resources but not others).

In this case, if IP addresses are manually free'd without actually deleting the handle, then subsequent calls to `ReleaseByHandle` will never delete the handle itself.

This makes sure that we delete the handle even if `ReleaseByHandle` didn't find any IPs to free.